tools/src/main/java/org/apache/kafka/tools/ShareConsumerPerformance.java

```diff
@@ -0,0 +1,327 @@
+                    joinTimeMsInSingleRound = new AtomicLong(0);
+                    lastReportTimeMs = currentTimeMs;
+                    lastMessagesRead = messagesRead;
+                    lastBytesRead = bytesRead;
+                }
+            }
+        }
+
+        if (messagesRead < numMessages)
+            System.out.printf("WARNING: Exiting before consuming the expected number of messages: timeout (%d ms) exceeded. " +
+                "You can use the --timeout option to increase the timeout.%n", recordFetchTimeoutMs);
+        totalMessagesRead.set(messagesRead);
+        totalBytesRead.set(bytesRead);
+    }
+
+    protected static class ShareConsumerPerfOptions extends CommandDefaultOptions {
+        private final OptionSpec<String> brokerListOpt;
+        private final OptionSpec<String> bootstrapServerOpt;
+        private final OptionSpec<String> topicOpt;
+        private final OptionSpec<String> groupIdOpt;
+        private final OptionSpec<Integer> fetchSizeOpt;
+        private final OptionSpec<Void> resetBeginningOffsetOpt;
+        private final OptionSpec<Integer> socketBufferSizeOpt;
+        private final OptionSpec<Integer> numThreadsOpt;
+        private final OptionSpec<Integer> numFetchersOpt;
+        private final OptionSpec<String> consumerConfigOpt;
+        private final OptionSpec<Void> printMetricsOpt;
+        private final OptionSpec<Void> showDetailedStatsOpt;
+        private final OptionSpec<Long> recordFetchTimeoutOpt;
+        private final OptionSpec<Long> numMessagesOpt;
+        private final OptionSpec<Long> reportingIntervalOpt;
+        private final OptionSpec<String> dateFormatOpt;
+        private final OptionSpec<Void> hideHeaderOpt;
+
+        public ShareConsumerPerfOptions(String[] args) {
+            super(args);
+            brokerListOpt = parser.accepts("broker-list", "DEPRECATED, use --bootstrap-server instead; ignored if --bootstrap-server is specified. The broker list string in the form HOST1:PORT1,HOST2:PORT2.")
+                .withRequiredArg()
+                .describedAs("broker-list")
+                .ofType(String.class);
+            bootstrapServerOpt = parser.accepts("bootstrap-server", "REQUIRED unless --broker-list(deprecated) is specified. The server(s) to connect to.")
+                .requiredUnless("broker-list")
+                .withRequiredArg()
+                .describedAs("server to connect to")
+                .ofType(String.class);
+            topicOpt = parser.accepts("topic", "REQUIRED: The topic to consume from.")
+                .withRequiredArg()
+                .describedAs("topic")
+                .ofType(String.class);
+            groupIdOpt = parser.accepts("group", "The group id to consume on.")
+                .withRequiredArg()
+                .describedAs("gid")
+                .defaultsTo("perf-share-consumer-" + RND.nextInt(100_000))
```

Comment:
I would default to `perf-share-consumer` because we will give preferential treatment to sharing in this way.